numpy/core/tests/test_scalar_methods.py

```diff
@@ -0,0 +1,93 @@
+"""
+Test the scalar constructors, which also do type-coercion
+"""
+from __future__ import division, absolute_import, print_function
+
+import fractions
+import pytest
+import numpy as np
+
+from numpy.testing import (
+    run_module_suite,
+    assert_equal, assert_almost_equal, assert_raises, assert_warns,
+    dec
+)
+
+class TestAsIntegerRatio(object):
+    # derived from the cpython test "test_floatasratio"
+
+    @pytest.mark.parametrize("ftype", [
+        np.half, np.single, np.double, np.longdouble])
+    @pytest.mark.parametrize("f, ratio", [
+        (0.875, (7, 8)),
+        (-0.875, (-7, 8)),
+        (0.0, (0, 1)),
+        (11.5, (23, 2)),
+        ])
+    def test_float_as_integer_ratio(self, ftype, f, ratio):
+        assert_equal(ftype(f).as_integer_ratio(), ratio)
+
+    @pytest.mark.parametrize("ftype, frac_vals, exp_vals", [
```

Comment:
Now that this is not random, I'm not sure the construction of this test makes much sense.

Perhaps simpler as:
```
@pytest.paramtrize('f', [
    np.half(...),
    np.single(...),
    np.double(...),
    np.longdouble("...")  # note string constructor needed
], id=lambda f: "{}-{}".format(f, f.dtype))
def test_roundtrip(f):
    if f.dtype == np.longdouble and not np.isfinite(f):
        pytest.skip("longdouble too small on this platform")
    n, d = f.as_integer_ratio()
    try:
        nf = np.longdouble(str(n))
        df = np.longdouble(str(d))
    except (OverflowError, RuntimeWarning):
        # the values may not fit in any float type
        continue
    assert nf / df == f
```

Reply:
I made all of your recent requested revisions, except for this one, which I only adopted partially. This version isn't any clearer to me, but feel free to just do this if you like.

I'm assuming you're hardcoding the `np.ldexp()` results instead of generating them in a loop.

Reply:
> I'm assuming you're hardcoding the np.ldexp() results instead of generating them in a loop.

Yes, exactly. If the values are arbitrary, showing computation steps is maybe not super useful.

> This version isn't any clearer to me

Fine to leave as is then